src/inference/src/ie_core.cpp

```diff
@@ -925,10 +931,10 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
     ov::InferencePlugin GetCPPPluginByName(const std::string& pluginName) const {
         OV_ITT_SCOPE(FIRST_INFERENCE, ie::itt::domains::IE_LT, "CoreImpl::GetCPPPluginByName");
 
-        std::lock_guard<std::mutex> lock(pluginsMutex);
         auto deviceName = pluginName;
         if (deviceName == ov::DEFAULT_DEVICE_NAME)
             deviceName = "AUTO";
+        std::lock_guard<std::mutex> lock(get_mutex(deviceName));
```

Comment:
Another thread2 may call 'RegisterPlugin' locking default mutex via 'get_mutex', and it will modify pluginRegistry there. And here another mutex is called allowing executing code below to read pluginRegistry. It is race condition

Reply:
Yes right thanks.